lagunita/lms/templates/instructor/instructor_dashboard_bottom.html

```diff
@@ -0,0 +1,8 @@
+<script language="JavaScript" type="text/javascript">
+  jQuery(document).ready(function () {
```

Comment:
We use jQuery all over the place so all calls to it can be modified to the short hand of:

`$(...`

Reply:
Also, convention is to not put any whitespace between the word function and it's parentheses, but to put one whitespace between the parentheses and the open brace. i.e. `function() {`

Reply:
Conversationally, is that an official convention, re: no space after `function`?

I've personally preferred to include the space.
My rationale has been that while the space is optional for anonymous functions,
it _is_ required for named functions.

So that way, mentally, I never have to think, "do i need a space here or not? what about a name".
Instead, I always insert the space and instead think, "do I need a name?"

Admittedly nit-picky, but that means if you give a name to an anonymous function
(which is often a good idea; it makes debug stack traces clear!),
the diff is just the addition of a name, not the space too :P

```js
function () {
}
```

becomes

```js
function foo() {
}
```

I know this way certainly isn't universal, but is the other form?

I was curious so I pulled some numbers on platform:

```
# find cms common lms openedx -name '*.js' -print0 | xargs -0 grep 'function (' | wc -l
4429

# find cms common lms openedx -name '*.js' -print0 | xargs -0 grep 'function(' | wc -l
25067
```

I'm fine w/ this as is, just curious :)

Reply:
I dunno man both Google and jQuery agree with me 😉 :
https://google.github.io/styleguide/jsguide.html
https://contribute.jquery.org/style-guide/js/

Reply:
To be fair, I'm not sure that quite addresses my question..
What I was looking for was whether or not what you were advocating for
was actually an accepted community standard, not whether or not there
were prominent examples.

I'm aware that people use that form; I acknowledged as much in my
comment. But I'm also aware that people _do_ use the other form as
well, both inside our codebase and out; that's why I'm asking about a
standard.

Pointing out an example of its use in the wild reads more like an
anecdote or in this case, name-dropping Google/JQ, an Appeal to
Authority (AtA).

That's problematic here because:
- Neither of those style guides _actually_ prescribe that syntax.
  For me, this lends weight to the idea that there really isn't any
  standardization of syntax here.  Style guides are prescriptive manuals
  for opinionated syntax. That this syntax is only casually included
  descriptively and not explicitly prescribed, leads me to believe that
  they aren't too opinionated here.
- Anecdotes and AtA are especially risky because it's easy to find
  conflicting examples:
  - Specifically here w/ jQuery and function declarations, while they
    are using _one_ bit of syntax that you may like, they're using another
    (on the same line!) that we definitely don't endorse:
    - they describe wrapping function arguments with spaces [2]:
      ```js
      function( foo, bar, baz ) {
      }
      ```
    - While I certainly won't dismiss jQuery out-of-hand, this does at
      least make me think, "maybe I shouldn't take their word as gospel,
      especially for function declarations".
  - Google's style guide expressly forbids use of the `var` keyword for
    variable declaration [1], a pattern we're not enforcing any time soon.
  - As a counter-example in the other direction, user `substack`, one
    of, if not _the_ most prolific JS package author in the history of
    the Node/JS ecosystem [3], includes the space everywhere.

Given this, the prominent counter-examples and lack of explicit
endorsement, I'm inclined to view this as _not_ standardized, and as
such, not consider it a blocker in the future.

- [1] https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations
- [2] https://github.com/jquery/jquery/blob/master/src/ajax.js#L197
- [3] https://www.npmjs.com/~substack
      Seriously, dude is wild; he has nearly 1000 packages/repos!

Reply:
Clearly you didn't see the emoji...It looks like either form is allowed.

Reply:
Clearly you've never seen me debate syntax and semantics ;)
 😃  😃  😃  😃  😃  😃  😃  😃  😃

Reply:
But real talk, I was genuinely curious if there was a standard or not;
now I know!